exporter/datadogexporter/internal/metadata/internal/gohai/gohai.go

```diff
@@ -0,0 +1,77 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gohai // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/metadata/internal/gohai"
+
+import (
+	"github.com/DataDog/gohai/cpu"
+	"github.com/DataDog/gohai/filesystem"
+	"github.com/DataDog/gohai/memory"
+	"github.com/DataDog/gohai/network"
+	"github.com/DataDog/gohai/platform"
+	"go.uber.org/zap"
+)
+
+// GetPayload builds a payload of every metadata collected with gohai except processes metadata.
+// Parts of this are based on datadog-agent code
+// https://github.com/DataDog/datadog-agent/blob/94a28d9cee3f1c886b3866e8208be5b2a8c2c217/pkg/metadata/internal/gohai/gohai.go#L27-L32
+func GetPayload(logger *zap.Logger) Payload {
```

Comment:
It's a bit uncommon (but not disallowed) to use the `Get*` prefix in Go: https://go.dev/doc/effective_go#Getters. I think this being a constructor, a better name is `NewPayload`.